proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package osmosis.cosmwasmpool.v1beta1;
+
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v15/x/cosmwasmpool/model";
+
+message CosmWasmPool {
+  option (gogoproto.goproto_getters) = false;
+  option (gogoproto.goproto_stringer) = false;
+  option (cosmos_proto.implements_interface) = "PoolI";
+  string pool_address = 1 [ (gogoproto.moretags) = "yaml:\"pool_address\"" ];
+  string contract_address = 2
+      [ (gogoproto.moretags) = "yaml:\"contract_address\"" ];
+  uint64 pool_id = 3;
+  uint64 code_id = 4;
```

Comment:
Do we need to store the code_id? It can be retrieved from the ContractInfo (using the addr) once the contract is instantiated.

Reply:
Doesn't storing it here prevent one more query in the required path though?

Reply:
> Doesn't storing it here prevent one more query in the required path though?

I think it does.

Also, I think we might want to store the code id for security purposes so that it is impossible to upgrade the contract to something malicious.

We can allow this to be upgradeable via gov prop. WDYT?

Going to make an issue to track this discussion

Reply:
Tracking discussion here: https://github.com/osmosis-labs/osmosis/issues/4779

Reply:
> Doesn't storing it here prevent one more query in the required path though?

Only when the code id is being used, which is only on instantiation (AFAICT). Not sure when else we would need the code id.

> Also, I think we might want to store the code id for security purposes so that it is impossible to upgrade the contract to something malicious.

We can make the contract non-upgradable by giving it no admin. 

Also if we wanted to enforce this by checking the code id, we would need to query the ContractInfo on every call to see that it matches what is stored.